examples/cfd/swe_nonlinear_pino/train_swe_nl_pino.py

```diff
@@ -0,0 +1,263 @@
+    swe_nl_node = swe_nl.make_nodes()
+
+    # Load in dataset and make dataloader
+    data = torch.load(
+        to_absolute_path("datasets/swe_nl_dataset.pt"),
+        map_location=torch.device("cpu"),
+        weights_only=True,
+    )
+    dataset = DataLoader2D_swe(
+        data, cfg.data.nx, cfg.data.nt, cfg.data.sub, cfg.data.sub_t
+    )
+
+    train_loader = dataset.make_loader(
+        cfg.data.n_train, cfg.batchsize, start=0, train=True
+    )
+    test_loader = dataset.make_loader(
+        cfg.data.n_test, cfg.batchsize, start=cfg.data.n_train, train=False
+    )
+
+    # Define FNO model architecture
+    model = FNO(
+        in_channels=cfg.model.fno.in_channels,
+        out_channels=cfg.model.fno.out_channels,
+        decoder_layers=cfg.model.fno.decoder_layers,
+        decoder_layer_size=cfg.model.fno.decoder_layer_size,
+        dimension=cfg.model.fno.dimension,
+        latent_channels=cfg.model.fno.latent_channels,
+        num_fno_layers=cfg.model.fno.num_fno_layers,
+        num_fno_modes=cfg.model.fno.num_fno_modes,
+        padding=cfg.model.fno.padding,
+    ).to(device)
+
+    optimizer = torch.optim.Adam(
+        model.parameters(), betas=(0.9, 0.999), lr=cfg.start_lr
+    )
+
+    scheduler = torch.optim.lr_scheduler.MultiStepLR(
+        optimizer, milestones=cfg.milestones, gamma=cfg.gamma
+    )
+
+    S, T = dataset.S, dataset.T
+    batch_size = cfg.batchsize
+    padding = cfg.model.fno.padding
+    nfields = cfg.model.fno.out_channels
+
+    for epoch in range(cfg.max_epochs):
+        # wrap epoch in launch logger for console logs
+        with LaunchLogger(
+            "train",
+            epoch=epoch,
+            num_mini_batch=len(train_loader),
+            epoch_alert_freq=10,
+        ) as log:
+            for x, y in train_loader:
+                # Pad input to be put into model
+                x, y = x.to("cuda"), y.to("cuda")
+                x_in = F.pad(x, (0, 0, 0, padding), "constant", 0)
+                x_in = x_in.permute(0, 4, 3, 1, 2)
+                # compute forward pass and unpad output
+                out = (
+                    model(x_in)
+                    .permute(0, 3, 4, 2, 1)
+                    .reshape(batch_size, S, S, T + padding, nfields)
+                )
+                out = out[..., :-padding, :]
+                s0 = x[..., 0, -1]
+
+                # Compute PDE loss using 'modulus' functions or method from 'original' paper
+                if cfg.loss.derivative == "original":
+                    loss_pde = PINO_loss_swe_nonlin(
+                        out,
+                        g=cfg.data.g,
+                        nu=cfg.data.nu,
+                        h_weight=cfg.loss.h_loss,
+                        u_weight=cfg.loss.u_loss,
+                        v_weight=cfg.loss.v_loss,
+                    )
+                elif cfg.loss.derivative == "modulus":
+                    loss_pde = modulus_fourier(
```

Comment:
nit: "nu" can be an input from the configuration here too?

Reply:
nu is already included in swe_nl_node when we define it, so i don't think we need it there again. is there another reason to have it there?

Reply:
Oh sorry, my bad- you are right!